drivers/hud_data_quality_report/app/models/hud_data_quality_report/generators/fy2020/base.rb

```diff
@@ -0,0 +1,1078 @@
+      end
+    end
+
+    private def clients_populated?
+      @report.report_cells.joins(universe_members: :dq_client).exists?
+    end
+
+    private def clients_with_enrollments(batch)
+      enrollment_scope.where(client_id: batch.map(&:id)).group_by(&:client_id)
+    end
+
+    private def enrollment_scope
+      preloads = {
+        enrollment: [
+          :client,
+          :disabilities,
+          :current_living_situations,
+          :project,
+          :services,
+          :income_benefits,
+          :income_benefits_at_exit,
+          :income_benefits_at_entry,
+          :income_benefits_annual_update,
+          :enrollment_coc_at_entry,
+          :health_and_dvs,
+          :exit,
+        ],
+      }
+      scope = GrdaWarehouse::ServiceHistoryEnrollment.
+        entry.
+        open_between(start_date: @report.start_date, end_date: @report.end_date).
+        joins(:enrollment).
+        preload(preloads)
+      scope = scope.in_project(@report.project_ids) if @report.project_ids.present? # for consistency with client_scope
+      scope
+    end
+
+    private def overlapping_enrollments(enrollments, last_enrollment)
+      last_enrollment_end = last_enrollment.last_date_in_program || Date.tomorrow
+      enrollments.select do |enrollment|
+        enrollment_end = enrollment.last_date_in_program || Date.tomorrow
+
+        enrollment.id != last_enrollment.id && # Don't include the last enrollment
+          enrollment.data_source_id == last_enrollment.data_source_id &&
+          enrollment.project_id == last_enrollment.project_id &&
+          enrollment.first_date_in_program < last_enrollment_end &&
+          enrollment_end > last_enrollment.first_date_in_program
+      end.map(&:enrollment_group_id).uniq
+    end
+
+    private def ages_for(household_id, date)
+      households[household_id].map { |client| GrdaWarehouse::Hud::Client.age(date: date, dob: client[:dob]) }
+    end
+
+    private def households
+      @households ||= {}.tap do |hh|
+        enrollment_scope.where(client_id: @generator.client_scope).find_each do |enrollment|
+          hh[enrollment.household_id] ||= []
+          hh[enrollment.household_id] << {
+            source_client_id: enrollment.enrollment.client.id,
+            dob: enrollment.enrollment.client.DOB,
+            veteran_status: enrollment.enrollment.client.VeteranStatus,
+            chronic_status: enrollment.enrollment.chronically_homeless_at_start?,
+            relationship_to_hoh: enrollment.enrollment.RelationshipToHoH,
+          }
+        end
+      end
+    end
+
+    private def household_member_data(enrollment)
+      # return nil unless enrollment[:head_of_household]
+
+      households[enrollment.household_id]
+    end
+
+    private def household_veterans_chronically_homeless?(dq_client)
+      adults = household_adults(dq_client)
+      veterans = household_veterans(adults)
+      household_chronically_homeless_clients(veterans).any?
+    end
```

Comment:
Should some of these be moved up one level to a shared concern?  I think we decided not to since HUD's updates don't come all at once, but that does make me worry.

Reply:
We did decide to keep them separate, but, maybe things that come out of the reporting glossary should be extracted as they intend those to be the same?